libutils/mutex.c

```diff
@@ -69,3 +68,53 @@ void __ThreadUnlock(pthread_mutex_t *mutex,
         exit(101);
     }
 }
+
+int __ThreadWait(pthread_cond_t *pcond, pthread_mutex_t *mutex, int timeout,
+                    const char *funcname, const char *filename, int lineno)
+{
+    int result = 0;
+
+    if (timeout == THREAD_BLOCK_INDEFINITELY)
+    {
+        result = pthread_cond_wait(pcond, mutex);
+    }
+    else if (timeout == THREAD_BLOCK_NONE)
+    {
+        return 0;
```

Comment:
This should still call `pthread_cond_timedwait()` and return a value saying if the condition was signalled or not.

Reply:
I'm not sure if I agree. That will add unnecessary calls if you want to run the function without waiting for a signal.

Also, I think the signal is only sent to threads that are waiting, which means that it will never be in a situation where it will return that it has been signalled. Correct me if I'm wrong!

Reply:
...I guess that option then becomes completely useless, just like not running the function at all. I think I will remove it and only use the `THREAD_BLOCK_INDEFINITELY`.